src/ServiceManager.php

```diff
@@ -78,29 +86,39 @@ class ServiceManager implements ServiceLocatorInterface
     /** @var ContainerInterface */
     protected $creationContext;
 
-    /** @var string[][]|Factory\DelegatorFactoryInterface[][] */
+    /**
+     * @var string[][]|Factory\DelegatorFactoryInterface[][]
+     * @psalm-var DelegatorsConfigurationType
+     */
     protected $delegators = [];
 
     /**
      * A list of factories (either as string name or callable)
      *
      * @var string[]|callable[]
+     * @psalm-var FactoriesConfigurationType
      */
     protected $factories = [];
 
-    /** @var Initializer\InitializerInterface[]|callable[] */
+    /**
+     * @var Initializer\InitializerInterface[]|callable[]
+     * @psalm-var InitializersConfigurationType
+     */
     protected $initializers = [];
 
-    /** @var array */
+    /**
+     * @var array
+     * @psalm-var LazyServicesConfigurationType
+     */
     protected $lazyServices = [];
 
-    /** @var null|Proxy\LazyServiceFactory */
+    /** @var null|LazyServiceFactory */
     private $lazyServicesDelegator;
 
     /**
      * A list of already loaded services (this act as a local cache)
      *
-     * @var array
+     * @var array<string,mixed>
```

Comment:
I've seen somewhere that only `array|object` is allowed for services. Maybe something to consider here?

Reply:
https://github.com/laminas/laminas-servicemanager/pull/98/files#diff-c88e8922673ece50d04aef21f4dc95b87bd465868eb073480467f87e626e3b0aR478

Reply:
but then we have to bc break `ServiceManager::doCreate` as it allows to return `mixed`